RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -520,6 +554,73 @@ ::computeWorstPFChargedIsolation(const T& photon, const U& pfCandidates,
 				 float dRvetoBarrel, float dRvetoEndcap, float ptMin){
 
 
+  float worstIsolation = 999;
+  std::vector<float> allIsolations;
+
+  float dRveto;
+  if (photon->isEB())
+    dRveto = dRvetoBarrel;
+  else
+    dRveto = dRvetoEndcap;
+
+  //Calculate isolation sum separately for each vertex
+  for(unsigned int ivtx=0; ivtx<vertices->size(); ++ivtx) {
+    
+    // Shift the photon according to the vertex
+    reco::VertexRef vtx(vertices, ivtx);
+    math::XYZVector photon_directionWrtVtx(photon->superCluster()->x() - vtx->x(),
+					   photon->superCluster()->y() - vtx->y(),
+					   photon->superCluster()->z() - vtx->z());
+    
+    float sum = 0;
+    // Loop over the PFCandidates
+    for(unsigned i=0; i<pfCandidates->size(); i++) {
+      
+      const auto& iCand = pfCandidates->ptrAt(i);
+
+      //require that PFCandidate is a charged hadron
+      reco::PFCandidate::ParticleType thisCandidateType = candidatePdgId(iCand, isAOD);
+      if (thisCandidateType != reco::PFCandidate::h) 
+	continue;
+
+      if (iCand->pt() < ptMin)
+	continue;
+      
+      float dxy=-999, dz=-999;
+      getImpactParameters(iCand, isAOD, *vtx, dxy, dz);
```

Comment:
this line third argument is the only difference between computeWorstPFChargedIsolationWithPVConstraint and computeWorstPFChargedIsolation.
Instead of copy-pasting the full implementation, please add an argument to computeWorstPFChargedIsolation which can pick ```pv``` or ```*vtx``` here